### PR TITLE
Milestone 3: Setup association

### PR DESCRIPTION
**Pull request by zdnahom**
## Milestone 3: Setup association
In this milestone I : 
- Created Classroom class
- Created Book class
- Created Rental class
- Implemented has-a and belongs-to the relationship between Classroom and Student class
- Implemented Many-to-many relationship between Book and Person class using an intermediate class (Rental)

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @parent_permission = parent_permission
     @age = age
+    @rentals = []
```
Comment (codecaiine) [inline person.rb:14]: - [ ] the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. Kindly check out the example below ⬇️  Please implement it in your code to make it completed 👍
```
 def add_rental(book, date)
    Rental.new(date, book, self)
  end

```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (codecaiine) [inline book.rb:16]: - [ ] The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. Please check out the example below ⬇️  Please implement it in your code to make it completed 👍 
 ```
def add_rental(person, date)
    Rental.new(date, self, person)
 end

```

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @zdnahom   , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Good class relationship setup :heavy_check_mark: 
- No rubocop errors :heavy_check_mark: 
- Good use of gitflow :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (zdnahom) [reply person.rb:14]: Thanks for your insights @codecaiine, I will implement it right now.

File: book.rb (hunk 2 above)
Comment (zdnahom) [reply book.rb:16]: Thanks for your insights @codecaiine, I will take care of it too 😉 .

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @zdnahom , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zdnahom** [comment]
Thanks for your review @VICTOR-CHUKWU 🙌

**Pull request merged**